controllers/provider-aws/pkg/webhook/controlplane/ensurer.go

```diff
@@ -205,3 +204,42 @@ func (e *ensurer) EnsureKubernetesGeneralConfiguration(ctx context.Context, data
 	*data = buf.String()
 	return nil
 }
+
+func appendUniqueUnit(units *[]extensionsv1alpha1.Unit, unit extensionsv1alpha1.Unit) {
+	for _, un := range *units {
+		if un.Name == unit.Name {
+			return
+		}
+	}
+	*units = append(*units, unit)
+}
+
+// EnsureAdditionalUnits ensures that additional required system units are added.
+func (e *ensurer) EnsureAdditionalUnits(ctx context.Context, units *[]extensionsv1alpha1.Unit) error {
+	var (
+		command              = "start"
+		trueVar              = true
+		customMTUUnitContent = `[Unit]
+    Description=Apply a custom MTU to eth0
+    Requires=eth0.network
+
+[Service]
+    Type=oneshot
+    RemainAfterExit=yes
+    ExecStart=/usr/bin/ip link set dev eth0 mtu 1460`
+	)
+
+	appendUniqueUnit(units, extensionsv1alpha1.Unit{
+		Name:    "custom-mtu.service",
+		Enable:  &trueVar,
+		Command: &command,
+		Content: &customMTUUnitContent,
+	})
+
+	return nil
+}
+
+// EnsureAdditionalFiles ensures that additional required system files are added.
+func (e *ensurer) EnsureAdditionalFiles(ctx context.Context, files *[]extensionsv1alpha1.File) error {
```

Comment:
See above, when implementing this in the `NoopEnsurer` you wouldn't need to implement it here.